kernel/Kconfig

```diff
@@ -1168,14 +1168,14 @@ config SMP_BOOT_DELAY
 config MP_NUM_CPUS
 	int "Number of CPUs/cores [DEPRECATED]"
 	default MP_MAX_NUM_CPUS
-	range 1 12
+	range 1 32
```

Comment:
Pedantry: this is technically a bisection problem, as you're increasing the range in kconfig before adding the code to handle it.  Properly this patch should be last.  I doubt anyone cares.